File: src/main/scala/org/openapitools/client/api/QrsItems.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QrsItems._

case class QrsItems (
  name: Option[String],
label: Option[String],
description: Option[String],
effectiveDate: Option[String],
untilDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
qrsType: Option[String],
links: Option[QrsItemsLinks])

object QrsItems {
  import DateTimeCodecs._

  implicit val QrsItemsCodecJson: CodecJson[QrsItems] = CodecJson.derive[QrsItems]
  implicit val QrsItemsDecoder: EntityDecoder[QrsItems] = jsonOf[QrsItems]
  implicit val QrsItemsEncoder: EntityEncoder[QrsItems] = jsonEncoderOf[QrsItems]
}

File: client/clojure/src/cdisc_library_api/specs/qrs_items.clj
(ns cdisc-library-api.specs.qrs-items
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.qrs-items-links :refer :all]
            )
  (:import (java.io File)))


(def qrs-items-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :untilDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :qrsType) string?
   (ds/opt :_links) qrs-items-links-spec
   })

(def qrs-items-spec
  (ds/spec
    {:name ::qrs-items
     :spec qrs-items-data}))
